src/Microsoft.SqlTools.SqlCore/ObjectExplorer/SmoModel/SmoColumnCustomNode.cs

```diff
@@ -212,12 +212,7 @@ internal static string GetTypeSpecifierLabel(DataType dataType, UserDefinedDataT
                         typeName += "(max)";
                         break;
                     case SqlDataType.Vector:
-                        // Temporary workaround to convert the maxLength to dimensions for vector types
-                        // until SMO is updated to store the actual dimensions of the vector type.
-                        // https://msdata.visualstudio.com/SQLToolsAndLibraries/_workitems/edit/3906463
-                        // dimensions = (length - 8) / 4
-                        // https://learn.microsoft.com/sql/t-sql/data-types/vector-data-type
-                        typeName += $"({(dataType.MaximumLength - 8) / 4})";
```

Comment:
With the change to `dataType.VectorDimensions` we no longer need to do any of these calculations to get the typeName?

Reply:
Correct.  Those calculations were a work-around in an earlier version because the engine sysviews didn't yet expose this properly, meaning SMO didn't have a property for them.  That's since solved.